applications/callflow/src/module/cf_resources.erl

```diff
@@ -341,3 +344,26 @@ handle_channel_destroy(<<"loopback", _/binary>>, _JObj) ->
     {<<"TRANSFER">>, 'ok'};
 handle_channel_destroy(_, JObj) ->
     {kz_call_event:hangup_cause(JObj), kz_call_event:hangup_code(JObj)}.
+
+-spec get_privacy_prefs(kapps_call:call()) -> kz_term:proplist().
+get_privacy_prefs(Call) ->
+    lager:debug("Checking inception of call"),
+    case kapps_call:inception(Call) of
+        'undefined' -> get_privacy_prefs_from_endpoint(Call);
+        _Else -> []
+    end.
+
+-spec get_privacy_prefs_from_endpoint(kapps_call:call()) -> kz_term:proplist().
+get_privacy_prefs_from_endpoint(Call) ->
+    lager:debug("Call is outbound, checking caller_id_outbound_privacy value"),
+    {'ok', Endpoint} = kz_endpoint:get(Call),
+    case kz_json:get_value([<<"caller_id_options">>, <<"outbound_privacy">>], Endpoint) of
+        'undefined' ->
+            [];
```

Comment:
what happens if cf_privacy is called but there's not caller_id_options for privacy ?

Reply:
or what happens if cf_privacy is called and there's calker_id_options for privacy ?

Reply:
This doesn't interact with cf_privacy. If there are no called_id_options for outbound_privacy then this module will not add any ccvs to the call.

Reply:
what if there are caller_id_options for outbound_privacy and cf_privacy was called before ? cf_privacy settings should override these

Reply:
when cf_privacy and outbound_privacy are both set, the resulting privacy will be the union of the two. If there are no outbound_privacy options, cf_privacy will have full control. If outbound_privacy is name only and cf_privacy is number only, both name and number will be hidden.

Reply:
my point exactly, this is not correct

Reply:
From my understanding you think that if one privacy option is set with cf_privacy, and the other is set with caller_id_options, then only the cf_privacy option should be used? I think it makes more sense that the user's decisions to hide caller ID should be respected and cf_privacy can be used to partially override some of the caller_id_options on a case-by-case basis.

Reply:
`caller_id_options` should be treated as defaults overridden by cf_privacy or set as an option in cf_privacy.
if `both` are set into caller_id_options and i want to send display name and/or number, how would cf_privacy help with that ?
this needs to be worked

Reply:
As far as I know, there is no way for cf_privacy to specify that you want to display everything, only hide which would mean you could not user cf_privacy to show your name and number if they are set to be hidden with the caller_id_options. However I can change it so that if cf_privacy and caller_id_options.outbound_privacy are both set, and are different to each other, then cf_privacy will be preferred completely instead of a merge.

Reply:
this PR introduced a change in behaviour, cf_privacy was mandatory and no longer is.  this will certainly cause issues and unnecessary support tickets. cf_privacy must be changed to either `merge` or `overwrite`, and cf_resources should respect cf_privacy if set or add caller_id_options if not set.
the default behaviour should be preserved, ie, default in cf_privacy should be `overwrite`